site/static/api/device-registry-v1.yaml

```diff
@@ -0,0 +1,724 @@
+                  application/json:
+                     schema:
+                        $ref: '#/components/schemas/Device'
+               headers:
+                  ETag:
+                     description: Version of the resource
+                     schema:
+                        type: string
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            404:
+               $ref: '#/components/responses/NotFound'
+
+      put:
+         tags:
+            - devices
+         summary: Update existing device registration
+         operationId: updateRegistration
+         parameters:
+            - $ref: '#/components/parameters/resourceVersion'
+         requestBody:
+            description: Updated device registration
+            content:
+               application/json:
+                  schema:
+                     $ref: '#/components/schemas/Device'
+            required: true
+         responses:
+            204:
+               $ref: '#/components/responses/Updated'
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            404:
+               $ref: '#/components/responses/NotFound'
+            412:
+               $ref: '#/components/responses/ResourceVersionMismatch'
+
+      delete:
+         tags:
+            - devices
+         summary: Delete device registration
+         operationId: deleteRegistration
+         parameters:
+            - $ref: '#/components/parameters/resourceVersion'
+         responses:
+            204:
+               $ref: '#/components/responses/Delete'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            404:
+               $ref: '#/components/responses/NotFound'
+            412:
+               $ref: '#/components/responses/ResourceVersionMismatch'
+
+   /credentials/{tenantName}/{deviceId}:
+
+      parameters:
+         - $ref: '#/components/parameters/tenantName'
+         - $ref: '#/components/parameters/deviceId'
+
+      get:
+         tags:
+            - credentials
+         summary: Get credentials set of a device.
+         description: |
+            Get the credentials set of a device. As long as the device is
+            registered and the user has read access to is, this call should
+            never return "not found".
+         operationId: getAllCredentials
+         responses:
+            200:
+               description: Operation successful
+               content:
+                  application/json:
+                     schema:
+                        $ref: '#/components/schemas/CredentialsSet'
+               headers:
+                  ETag:
```

Comment:
this needs to be marked as *optional*, doesn't it?

Reply:
From the OpenAPI documentation:

> By default, OpenAPI treats all request parameters as optional. You can add required: true to mark a parameter as required. Note that path parameters must have required: true, because they are always required.